src/data/list/big_operators.lean

```diff
@@ -193,15 +193,40 @@ lemma _root_.commute.list_sum_left [non_unital_non_assoc_semiring R] (b : R) (l
   commute l.sum b :=
 (commute.list_sum_right _ _ $ λ x hx, (h _ hx).symm).symm
 
+@[to_additive sum_le_sum] lemma forall₂.prod_le_prod' [preorder M]
+  [covariant_class M M (function.swap (*)) (≤)] [covariant_class M M (*) (≤)]
+  {l₁ l₂ : list M} (h : forall₂ (≤) l₁ l₂) : l₁.prod ≤ l₂.prod :=
+begin
+  induction h with a b la lb hab ih ih',
+  { refl },
+  { simpa only [prod_cons] using mul_le_mul' hab ih' }
+end
+
+@[to_additive sum_le_sum] lemma sublist.prod_le_prod' [preorder M]
+  [covariant_class M M (function.swap (*)) (≤)] [covariant_class M M (*) (≤)]
+  {l₁ l₂ : list M} (h : l₁ <+ l₂) (h₁ : ∀ a ∈ l₂, (1 : M) ≤ a) : l₁.prod ≤ l₂.prod :=
```

Comment:
Should this be the stronger
```suggestion
  {l₁ l₂ : list M} (h : l₁ <+ l₂) (h₁ : ∀ a ∈ l₂.diff l₁, (1 : M) ≤ a) : l₁.prod ≤ l₂.prod :=
```

Reply:
Do you have a proof? It seems that we lack a lot of supporting lemmas about `diff` and `sublist`.

Reply:
Anyway, I suggest that we leave these lemmas as is, then add `diff` versions as needed. E.g., these proofs don't need `classical.choice` while a proof based on the `diff` version will need this axiom to add `decidable_eq M`.

Reply:
Mind adding a brief comment to the docstring saying "this could be made stronger by choosing `h₁ : ∀ a ∈ l₂.diff l₁, (1 : M) ≤ a`" or similar? Otherwise I agree it's fine as is.

Reply:
Done.